src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -328,6 +328,8 @@ public NativeMemoryAllocation get(NativeMemoryEntryContext<?> nativeMemoryEntryC
 
             // Cache Miss
             // Evict before put
+            // preload the graph file before proceeding to load the graph into memory
```

Comment:
It's not preloading the graph file, isn't it? 🤔 
Opening a stream != loading the graph file, I believe